archeryutils/constants.py

```diff
@@ -1,2 +1,87 @@
 """Constants used in the archeryutils package."""
-YARD_TO_METRE = 0.9144
+
+_CONVERSIONS = {
+    "metre": 1.0,
+    "yard": 0.9144,
+    "cm": 0.01,
+    "inch": 0.0254,
+}
+
+_YARD_ALIASES = {
+    "Yard",
+    "yard",
+    "Yards",
+    "yards",
+    "Y",
+    "y",
+    "Yd",
+    "yd",
+    "Yds",
+    "yds",
+}
+
+_METRE_ALIASES = {
+    "Metre",
+    "metre",
+    "Metres",
+    "metres",
+    "M",
+    "m",
+    "Ms",
+    "ms",
+}
+
+_CM_ALIASES = {
+    "Centimeter",
+    "centimeter",
+    "Centimeters",
+    "centimeters",
+    "CM",
+    "cm",
+    "CMs",
+    "cms",
+}
+
+_INCH_ALIASES = {
+    "Inch",
+    "inch",
+    "Inches",
+    "inches",
+}
+
+_ALIASES = dict(
+    yard = _YARD_ALIASES,
+    metre = _METRE_ALIASES,
+    cm = _CM_ALIASES,
+    inch = _INCH_ALIASES,
+)
+
+class Length:
+    yard = _YARD_ALIASES
+    metre = _METRE_ALIASES
+    cm = _CM_ALIASES
+    inch = _INCH_ALIASES
+
+    _reversed = {
+        alias: name
+        for name in _CONVERSIONS
+        for alias in _ALIASES[name]
+    }
+
+    _conversions = {
+        alias: factor
+        for name, factor in _CONVERSIONS.items()
+        for alias in _ALIASES[name]
+    }
+
+    @classmethod
+    def to_metres(cls, value: float, unit: str) -> float:
+        return cls._conversions[unit] * value
+
+    @classmethod
+    def from_metres(cls, metre_value: float, unit: str) -> float:
+        return metre_value / cls._conversions[unit]
+
+    @classmethod
+    def definitive_name(cls, alias: str) -> str:
+        return cls._reversed[alias]
```

Comment:
~Rather than 'definitive' perhaps consider 'native_units' or 'original_units'? - one could argue that the 'definitive' unit of distance in this code is metres since everything is converted to that before applying equations.~

After figuring out how this works I wonder if it is neater to preserve the original functionality and just force any 'alternative' input format to be standardised to 'metres', 'yards' etc. This would be more consistent throughout and simplify the logic.

I would still consider a rename to `definitive_units` rather than `definitive_name`.

Reply:
`definitive_units` is a better method name I agree.

This PR still forces standardisation of any input to the definitve names, it just includes the aliases in the conversion dictionaries as well I just found that the coercion to a definitive name fitted more neatly factored into the length class rather than in the init functions where it was used.

Reply:
Ah, yes, I misinterpreted what was happening with the `_reversed` variable and `definitive` method when reading previously. #docstrings